blog/content/post/2021-10-03-fobnail-promotion.md

```diff
@@ -0,0 +1,266 @@
+influence Fobnail as it inspects the attestations made by the system. The
+architecture is based on the IEFT specification - Remote ATtestation
+ProcedureS ([RATS](https://datatracker.ietf.org/doc/draft-ietf-rats-architecture/))
+and the attestation procedure on Reference Interaction Model for
+Challenge-Response-based Remote Attestation ([CHARRA](https://tools.ietf.org/id/draft-birkholz-rats-reference-interaction-model-00.html)).
+
+So in short, the main principles of Fobnail are:
+
+- to make the attestation as simple as possible (e.g. by using LED indicators
+  to inform a user about the decision of trustworthiness)
+- act as the axiomatically trusted device (an [iTurtle](https://www.usenix.org/legacy/event/hotsec07/tech/full_papers/mccune/mccune_html/index.html))
+  for the attestation process
+- provide attestation services without network connection
+
+So how does it improve situation for the mentioned projects? As long as the
+firmware is protected the risk is not so high that the components have been
+tampered with. But the attestation process over the network may be hijacked or
+return untrustworthy result. By using fobnail, whole attestation and decision
+process is moved to the tiny USB device so that the user can confirm the result
+visually. This solves the problem of untrusted networks or attestation servers.
+
+Secondly if we do not trust that the firmware is bug-free and provides reliable
+measurements one may use D-RTM when booting the target Linux kernel and perform
+the attestation based on D-TM measurements with Fobnail.
+
+Thirdly when we look at TOTP/HOTP the whole decision process of secret
+unsealing is done on the platform, also the platform is responsible to display
+or send the 6 digit code to the display or USB token. This creates a potential
+risk that the code has been faked and the measurements aren't really reliable.
+Fobnail removes such risk by moving the platform state evaluation on the token
+side. Before the Fobnail token becomes usable it must be provisioned first with:
+
+- Reference Integrity Measurement (RIM) Database (approved set of platform
+  measurements)
+- Attestation Policy Database (decision policies used to evaluate the
+  trustworthiness of the attestation data)
+- Attestation Protected Object (optional secret data that is going to be
+  unsealed on successful attestation process)
+- Identity and encryption certificates
+
+So Fobnail knows about all the known good platform states. The host
+responsibility in the attestation process is to provide the TPM quote (set of
+platform integrity measurements, event logs, etc.) signed by the attestation
+key to the token for evaluation. Then the decision is returned to the host and
+indicated to the platform owner with a physical or digital response. Fobnail is
+also a flexible device which allows to perform the attestation of multiple
+devices (by using multiple RIMs and policies).
```

Comment:
@miczyg1 I'm little bit worried we talking about  Fobnail as a device, we should change that narrative to not confuse USB token early adopters. Fobnail is set of features and architecture related with attestation server on separate device accessible without Internet or Intranet connection. Fobanil main target for attestation are D-RTM PCRs and we can say this is other side of equation, but that decision do not limit Fobnail features compatibilities and possibilities.

Reply:
Understood. Rephrasing it is pretty easy, but on the other hand, we have many places where Fobnail is advertised as a device/USB token, e.g.:
- line 67 of this post `At 3mdeb we aim to create an open hardware USB security token called [Fobnail]`
- fobnail.3mdeb.com `The Fobnail Token is a tiny open-source hardware USB device`
What we want Fobnail to be is not what we have claimed it to be in public.

Reply:
Rephrased the Fobnail to be an architecture in this context

Reply:
@miczyg1 let's adress also changes in public Fobnail documentation. I believe initially we wanted to create OSHW token with OSF, but now we rather going into direction of integrating on top of existing tokens with priority for early adopters. If used hardware will be OSHW created by us this is different story, most probably it would not be created by us, but still can be OSHW as most reasonable USB tokens.